### PR TITLE
Fix various moves

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

File: sim/battle.ts [hunk 1]
```diff
@@ -2471,6 +2471,8 @@ export class Battle {
 			// take priority from the base move, so abilities like Prankster only apply once
 			// (instead of compounding every time `getActionSpeed` is called)
 			let priority = this.dex.getMove(move.id).priority;
+			// Grassy Glide priority
+			if (move.onModifyPriority) priority = move.onModifyPriority.call(this, priority);
```
Comment (Zarel) [inline sim/battle.ts:2475]:
```suggestion
			priority = this.singleEvent('ModifyPriority', move, null, action.pokemon);
```

**Comment by KrisXV** [comment]
done @Zarel

**Comment by Zarel** [comment]
Thanks!

File: data/moves.ts [hunk 2]
```diff
@@ -13716,11 +13713,9 @@ export const BattleMovedex: {[moveid: string]: MoveData} = {
 		pp: 5,
 		priority: 0,
 		flags: {protect: 1, mirror: 1},
-		onPrepareHit(target, source, move) {
+		onTryHit(target, source, move) {
 			if (source.ignoringItem()) return false;
-			const item = source.getItem();
-			if (!this.singleEvent('TakeItem', item, source.itemData, source, source, move, item)) return false;
-			if (!item.id) return false;
+			if (!target.item) return false;
```
Comment (urkerab) [inline data/moves.ts:13718]: The description is no longer consistent with this change. (I guess you just need the shortDesc now?)

File: data/scripts.ts [hunk 3]
```diff
@@ -513,7 +513,7 @@ export const BattleScripts: BattleScriptsData = {
 		if (move.breaksProtect) {
 			for (const target of targets) {
 				let broke = false;
-				for (const effectid of ['banefulbunker', 'kingsshield', 'protect', 'spikyshield']) {
+				for (const effectid of ['banefulbunker', 'kingsshield', 'obstruct', 'protect', 'spikyshield']) {
```
Comment (urkerab) [inline data/scripts.ts:516]: (Wow, how long has that been broken?)

**Pull request merged**